File: An2/MPP/mpp-proiect-csharp-DeepSmeag/repo/InscriereRepo.cs
using mpp_proiect_csharp_DeepSmeag.domain;
using mpp_proiect_csharp_DeepSmeag.repo.interfaces;
using mpp_proiect_csharp_DeepSmeag.repo.utils;
using Npgsql;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace mpp_proiect_csharp_DeepSmeag.repo
{
    internal class InscriereRepo : IInscriereRepoInterface
    {
        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
        public bool deleteEntity(int idParticipant, int idProba)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "DELETE FROM \"Inscrieri\" WHERE \"idParticipant\"=@idParticipant AND \"idProba\"=@idProba";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idParticipant", idParticipant);
                sqlCommand.Parameters.AddWithValue("idProba", idProba);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }
        public bool deleteEntity(int id)
        {
            return false;
        }

        public List<Inscriere>? getAll()
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"Inscrieri\"";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                using var reader = sqlCommand.ExecuteReader();
                var inscrieri = new List<Inscriere>();
                while (reader.Read())
                {
                    var inscriere = new Inscriere(reader.GetInt32(0), reader.GetInt32(1));
                    inscrieri.Add(inscriere);
                }
                con.Close();
                return inscrieri;
            }
            catch (Exception e)
            {
                return null;
            }
        }

        public Inscriere? getInscriereByParticipantAndProba(int idParticipant, int idProba)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"Inscrieri\" WHERE \"idParticipant\"=@idParticipant AND \"idProba\"=@idProba";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idParticipant", idParticipant);
                sqlCommand.Parameters.AddWithValue("idProba", idProba);
                using var reader = sqlCommand.ExecuteReader();
                if (reader.Read())
                {
                    var inscriere = new Inscriere(reader.GetInt32(0), reader.GetInt32(1));
                    con.Close();
                    return inscriere;
                }
                con.Close();
                return null;
            }
            catch (Exception e)
            {
                return null;
            }
        }

        public Inscriere? getOne(int entityId)
        {
            return null;
        }

        public bool saveEntity(Inscriere entityType)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "INSERT INTO \"Inscrieri\" VALUES (@idParticipant, @idProba)";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idParticipant", entityType.IdParticipant);
                sqlCommand.Parameters.AddWithValue("idProba", entityType.IdProba);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }

        public bool updateEntity(Inscriere entityType)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "UPDATE \"Inscrieri\" SET \"idParticipant\"=@idParticipant, \"idProba\"=@idProba WHERE \"idParticipant\"=@idParticipant AND \"idProba\"=@idProba";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idParticipant", entityType.IdParticipant);
                sqlCommand.Parameters.AddWithValue("idProba", entityType.IdProba);
                sqlCommand.Prepare();
                sqlCommand.ExecuteNonQuery();
                con.Close();
                return true;
            }
            catch (Exception e)
            {
                return false;
            }
        }

        internal int getNumRegistrations(int idProba)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT COUNT(\"idParticipant\") FROM \"Inscrieri\" WHERE \"idProba\"=@idProba";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idProba", idProba);
                sqlCommand.Prepare();
                using var reader = sqlCommand.ExecuteReader();
                reader.Read();
                int num = reader.GetInt32(0);

                con.Close();
                return num;
            }
            catch (Exception e)
            {
                return -1;
            }
        }

        internal List<Inscriere> getInscrieriByParticipantId(int idParticipant)
        {
            try
            {
                using var con = DBUtils.GetNewConnection();
                con.Open();
                var sqlString = "SELECT * FROM \"Inscrieri\" WHERE \"idParticipant\"=@idParticipant";
                using var sqlCommand = new NpgsqlCommand(sqlString, con);
                sqlCommand.Parameters.AddWithValue("idParticipant", idParticipant);
                using var reader = sqlCommand.ExecuteReader();
                var inscrieri = new List<Inscriere>();
                while (reader.Read())
                {
                    var inscriere = new Inscriere(reader.GetInt32(0), reader.GetInt32(1));
                    inscrieri.Add(inscriere);
                }
                con.Close();
                return inscrieri;
            }
            catch (Exception e)
            {
                log.Error(e);
                return null;
            }
        }
    }
}

File: An2/MPP/mpp-proiect-java-DeepSmeag/persistenceModule/src/main/java/contests/persistence/InscriereRepo.java
package contests.persistence;

import contests.model.Inscriere;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import contests.persistence.interfaces.InscriereRepoInterface;
import contests.persistence.utils.DBConnection;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class InscriereRepo implements InscriereRepoInterface {
    private static final Logger logger = LogManager.getLogger();
    @Override
    public List<Inscriere> getAll() {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"Inscrieri\"")) {
            ResultSet resultSet = ps.executeQuery();
            ArrayList<Inscriere> inscrieri = new ArrayList<>();
            while (resultSet.next()) {
                int idParticipant = resultSet.getInt("idParticipant");
                int idProba = resultSet.getInt("idProba");
                Inscriere inscriere = new Inscriere(idParticipant, idProba);
                inscrieri.add(inscriere);
            }
            return inscrieri;
        } catch (SQLException e) {
            return null;

        }
    }

    @Override
    public Inscriere getOne(Integer integer) {
        return null;
    }

    @Override
    public boolean saveEntity(Inscriere entity) {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("INSERT INTO \"Inscrieri\" VALUES (?,?)")) {
            ps.setInt(1, entity.getIdParticipant());
            ps.setInt(2, entity.getIdProba());
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    @Override
    public boolean deleteEntity(Integer integer) {
        return false;
    }


    public boolean deleteEntity(int idParticipant, int idProba) {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("DELETE FROM \"Inscrieri\" WHERE idParticipant=? AND idProba=?")) {
            ps.setInt(0, idParticipant);
            ps.setInt(1, idProba);
            ps.executeUpdate();
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    @Override
    public boolean updateEntity(Inscriere entity) {
        return false;
    }

    @Override
    public Inscriere getInscriereByParticipantAndProba(int idParticipant, int idProba) {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"Inscrieri\" WHERE \"idParticipant\" = ? AND \"idProba\"= ? ")) {
            ps.setInt(0, idParticipant);
            ps.setInt(1, idProba);
            ResultSet resultSet = ps.executeQuery();


            Inscriere inscriere = new Inscriere(idParticipant, idProba);
            return inscriere;
        } catch (SQLException e) {
            return null;
        }
    }

    public int getNumRegistrations(Integer idProba) {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("SELECT COUNT(\"idParticipant\") FROM \"Inscrieri\" WHERE \"idProba\"= " + String.valueOf(idProba))) {
//            ps.setInt(0, idProba.intValue());
            ResultSet resultSet = ps.executeQuery();

            resultSet.next();
            return resultSet.getInt(1);
        } catch (SQLException e) {
            logger.error(e.getMessage(), e);
            return -1;
        }
    }

    public List<Inscriere> getInscrieriByParticipantId(Integer idParticipant) {
        try (Connection connection = DBConnection.getNewConnection();
             PreparedStatement ps = connection.prepareStatement("SELECT * FROM \"Inscrieri\" WHERE \"idParticipant\"= ?")){
            ps.setInt(1, idParticipant.intValue());
            ResultSet resultSet = ps.executeQuery();
            List<Inscriere> inscrieri = new ArrayList<>();
            while(resultSet.next()){
                int idProba = resultSet.getInt("idProba");
                Inscriere inscriere = new Inscriere(idParticipant, idProba);
                inscrieri.add(inscriere);
            }
            return inscrieri;
        } catch (SQLException e) {
            logger.error(e.getMessage(), e);
            return null;
        }
    }
}
